client/components/theme-type-badge/tooltip.tsx

```diff
@@ -0,0 +1,177 @@
+import { recordTracksEvent } from '@automattic/calypso-analytics';
+import {
+	DOT_ORG_THEME,
+	MARKETPLACE_THEME,
+	PREMIUM_THEME,
+	WOOCOMMERCE_THEME,
+} from '@automattic/design-picker';
+import { Button as LinkButton } from '@wordpress/components';
+import { createInterpolateElement } from '@wordpress/element';
+import { useTranslate } from 'i18n-calypso';
+import { useEffect } from 'react';
+import { useSelector } from 'calypso/state';
+import {
+	canUseTheme,
+	getThemeType,
+	isThemePurchased,
+	isMarketplaceThemeSubscribed,
+	getMarketplaceThemeSubscriptionPrices,
+} from 'calypso/state/themes/selectors';
+import getSelectedSiteId from 'calypso/state/ui/selectors/get-selected-site-id';
+import getSelectedSiteSlug from 'calypso/state/ui/selectors/get-selected-site-slug';
+
+interface Props {
+	themeId: string;
+}
+
+const ThemeTypeBadgeTooltip = ( { themeId }: Props ) => {
```

Comment:
Maybe we can refactor the link responsibility outside of the actual tooltip. We could create a ThemeLinkButton like so: 
```
const ThemeLinkButton = ({plan, children}) => {
    const siteSlug = useSelector(getSelectedSiteSlug);

    const goToCheckout = (plan) => {
        recordTracksEvent('calypso_theme_tooltip_upgrade_nudge_click', { plan });

        if (siteSlug) {
            const params = new URLSearchParams();
            params.append('redirect_to', window.location.href.replace(window.location.origin, ''));

            window.location.href = `/checkout/${encodeURIComponent(siteSlug)}/${plan}?${params.toString()}`;
        }
    };

    return <LinkButton isLink onClick={() => goToCheckout(plan)}>{children}</LinkButton>
};
```

Then we could remove that logic from the ThemeBadgeTooltip component and replace the message assignments with:
```
message = createInterpolateElement(
    translate(
        'This community theme can only be installed if you have the <Link>Business plan</Link> or higher on your site.'
    ),
    {
        Link: <ThemeLinkButton plan="business" />,
    }
);
```

Reply:
> Maybe we can refactor the link responsibility outside of the actual tooltip

Makes sense! However, since the logic of this changes a lot in a follow-up PR (https://github.com/Automattic/wp-calypso/pull/77675), I'll do that there.

Reply:
https://github.com/Automattic/wp-calypso/pull/77675/commits/f0ebbe40c39c7939ddc12d13dd7fa09f2cdba03d